### DISCUSSION TITLE
Add Overtime to next Schedule Item Start Time

### DISCUSSION THREAD

**Discussion by ElyFura** [cpvalente/ontime#1791] [Ideas]
Hello,

For us to use ontime theres a important feature missing which is the option to add overtime from a previous event to the start time of the next event.

For Example:

 ### Event 1
- Event 1 Start Time: 12:00:00
- Event 1 Planned End: 12:10:00
- Event 2 Planned Start: 12:15:00
- Event 1 Actuall End Time: 12:13:00

Would mean the Start of Event 2 should move to 12:18:00 including set up 5 Minute Delay between the two events.

Is there a way to implement this? it would be the last piece to make the Software Perfect for us.

**Comment by cpvalente** [comment]
Hi @ElyFuram, thank you for taking the time to make this proposal.

I believe we would need some more context as to what you work on and what you are trying to achieve.

1. Are you trying to move the schedule as shown in the backstage views? 
2. in the example you gave, what should happen to the time added? 
   - should it be compensated by making the event shorter
   - should it propagate through your rundown?

If it is easier, you can drop into our [discord server](https://discord.com/invite/eje3CSUEXm) and we can have a chat there

**Comment by ElyFura** [comment]
Hi @cpvalente,

For Our usecase we have only a fixed time for each event planned but it can change, means the event can get shorter or longer. The ideal would be if the public schedule would update accordingly to the actual times to use them on displays.

With the Example from above if the Event looks like this:

![image](https://github.com/user-attachments/assets/5bd7d3db-73f5-48ad-a861-e51f07f274d6)

From the Screen if the First Event ends 12:13 instead of 12:10 should the next event move its Start time from 12:10 to 12:13 with the 10 minute break between, means it should be shown as 12:23 instead of 12:20.

![image](https://github.com/user-attachments/assets/27566251-421f-40ed-8996-4a31df42d482)

**Comment by alex-Arc** [comment]
Hey @ElyFura 
I think these are some functions that might be better implemented with a control layer
a similar request has been made here https://github.com/bitfocus/companion-module-getontime-ontime/issues/85
I will try to remember getting back to you when we have this working

**Comment by ElyFura** [comment]
@alex-Arc thanks alot im looking forwars to it.

**Comment by alex-Arc** [comment]
if you want to try it out, the is a build of the companion module here https://github.com/alex-Arc/companion-module-getontime-ontime/actions/runs/12320262742

otherwise you will have to wait  for the next companion release

**Comment by cpvalente** [comment]
> important feature missing which is the option to add overtime from a previous event to the start time of the next event.

Hi @ElyFura , I apologise for the slow replies

I think this behaviour is already supported by Ontime, ie: not in the way that you mention
Ontime keeps track of the expected start time for all events. We call this projected time

You can turn on "show projected times" in the backstage view and the schedule should update to reflect the runtime changes.
Can you confirm whether this does what you need? is there anything missing?

**Comment by ElyFura** [comment]
> You can turn on "show projected times" in the backstage view and the schedule should update to reflect the runtime changes.

Hello @cpvalente ,

I looked at the option but its sadly not the right for us. In our use case for events we would need to change the start times of the shedule.

In our usecase shedule entires have just an estimated duration, means it can be under or over estimate and the rest of the shedule should apply to it.

For Example between each Event i a 10 minute setup break, means if the first event with a duration of 30 minutes starts at 12:00 and ends as planned on 12:30 nothing needs to change, the next event start at 12:40 as planned.

But lets say the first events has overtime and ends on 12:50 then the next event should set its start time after switching to it to 13:00.


I hope i could explain it better this time, basicly the start time would have to change dynamicly to the new times when you switch to the next event in the shedule.
